packages/go/analysis/ad/esc3.go

```diff
@@ -39,9 +39,17 @@ func PostADCSESC3(ctx context.Context, tx graph.Transaction, outC chan<- analysi
 		return nil
 	} else if collected, err := eca2.Properties.Get(ad.EnrollmentAgentRestrictionsCollected.String()).Bool(); err != nil {
 		return fmt.Errorf("error getting enrollmentagentcollected for eca2 %d: %w", eca2.ID, err)
-	} else if hasRestrictions, err := eca2.Properties.Get(ad.HasEnrollmentAgentRestrictions.String()).Bool(); err != nil {
-		return fmt.Errorf("error getting hasenrollmentagentrestrictions for ca %d: %w", eca2.ID, err)
 	} else {
+		// Assuming no enrollement agent restrictions if not collected
+		eARestrictions := false
+		if collected {
+			if hasRestrictions, err := eca2.Properties.Get(ad.HasEnrollmentAgentRestrictions.String()).Bool(); err != nil {
+				return fmt.Errorf("error getting hasenrollmentagentrestrictions for ca %d: %w", eca2.ID, err)
+			} else {
+				eARestrictions := hasRestrictions
```

Comment:
This needs to be `=` instead of `:=` because of var scoping. You're reassigning eaRestrictions to a new variable in this scope so its not affecting the original declaration